doc/api/next_api_changes/deprecations/24730-TH.rst

```diff
@@ -0,0 +1,24 @@
+rcParams type
+~~~~~~~~~~~~~
+Relying on ``rcParams`` being a ``dict`` subclass is deprecated.
+
+Nothing will change for regular users because ``rcParams`` will continue to
+be dict-like (technically fulfill the ``MutableMapping`` interface).
+
+However, there are some rare known cases that use
+``dict.__getitem__(rcParams, key)`` and
+``dict.__setitem__(rcParams, key, val)`` for direct data access that
+circumvents any validation logic. While this is technically public, it
+depends on the implementation detail that ``rcParams`` is currently a ``dict``
+subclass.
+
+Matplotlib 3.7 introduces ``rcParams._set(key, val)`` and
+``rcParams._get(key)`` as a replacements for these cases. They are
+intentionally marked private because there is very little need for external
+users to use them. However, if you have a compelling reason for direct data
+access, please switch from the dict functions to the new ``_get()`` and
+``_set()``. Even though marked private, we guarantee API stability for these
+methods, i.e., they are subject to Matplotlib's API and deprecation policy.
```

Comment:
```suggestion
Matplotlib 3.7 introduces ``rcParams._set(key, val)`` and
``rcParams._get(key)`` as a replacement to calling the parent methods. They are
intentionally marked private because there is little need for external
users to use them. However, if you have a compelling reason for direct `.RcParam` data
access, please switch from the dict functions to the new ``_get()`` and
``_set()``. Even though marked private, we guarantee API stability for these
methods and they are subject to Matplotlib's API and deprecation policy.
```